openproblems/data/tabula_muris_senis.py

```diff
@@ -73,20 +58,31 @@ def load_tabula_muris_senis(test=False, method_list=None, organ_list=None):
     and droplet-fat anndata sets. (no facs-fat dataset available)
     """
 
-    # df containing figshare links, method of collection, and organ for each
-    # tabula muris dataset
-    url_df = pd.read_csv(
-        os.path.join(
-            os.path.dirname(__file__),
-            "tabula_muris_senis_data_objects",
-            "tabula_muris_senis_data_objects.csv",
-        ),
-        header=0,
-    )
+    if not method_list:
+        raise ValueError("`method_list' argument not provided")
+    unknown_methods = set(method_list) - set(["facs", "droplet"])
+    if unknown_methods:
+        raise ValueError(
+            f"Unknown methods provided in `method_list': {','.join(unknown_methods)}. "
+            "Known methods are `facs' and `droplet'"
+        )
+    organ_list = [x.lower() for x in organ_list]
+
+    datasets_path = f"/curation/v1/collections/{COLLECTION_ID}"
+    url = f"{API_BASE}{datasets_path}"
+    res = requests.get(url=url)
+    datasets = res.json()["datasets"]
+
+    adata_list = []
+    for dataset in datasets:
+        if matching_dataset(dataset, method_list, organ_list):
+            adata_list.append(load_raw_counts(dataset))
```

Comment:
Do we want to check that all requested organs exist?

Reply:
Added this check in a separate function